### PR TITLE
[#15249] Input tests

### LINKED ISSUE #15249: Add tests for new input component
While solving #15025 it was noticed that test for that component are quite large to add them in its PR https://github.com/status-im/status-mobile/pull/15237.

This issue is about adding the missing tests for the input component.

### PR DESCRIPTION
**Pull request by ulisesmac**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

fixes #15249 

### Summary

[comment]: # (Summarise the problem and how the pull request solves it)
This PR adds the pending tests for the input component.

### Steps to test

Run component tests

status: ready

### CODE REVIEW

File: src/test_helpers/component.cljs [hunk 1]
```diff
@@ -71,3 +72,7 @@
 (defn is-null
   [element]
   (.toBeNull (js/expect element)))
+
+(defn was-called
```
Comment (ulisesmac) [inline src/test_helpers/component.cljs:76]: What do you think about this helper function added?

File: src/quo2/components/inputs/input/style.cljs [hunk 2]
```diff
@@ -83,15 +83,15 @@
...
-  (merge (text/text-style {:size :paragraph-1 :weight :regular})
-         {:flex                1
-          :text-align-vertical :top
-          :padding-right       0
-          :padding-left        (if small? 4 8)
-          :padding-vertical    (if small? 4 8)
-          :color               (:text colors-by-status)}
-         (when-not multiple-lines?
-           {:height (if small? 30 38)})))
+  (let [base-props (assoc (text/text-style {:size :paragraph-1 :weight :regular})
```
Comment (ulisesmac) [inline src/quo2/components/inputs/input/style.cljs:86]: There was a problem I recently noticed on the input component.

On devices with high densitiy of pixels and small font size set by the user, the text wrote inside this input was not centered, it was due to the `:text-align-vertical` added to all the input variants. Now it's only added when the input is `multiline` as suggested by the [React Native docs](https://reactnative.dev/docs/0.63/textinput#multiline).

Extra info: I used a physical Galaxy Z Fold 4 to notice it, previously I didn't see this problem.

**Comment by ulisesmac** [comment]
I'm testing the main render variants since there're too many possible combinations.
If you think an important test was skipped and should be included, please let me know :smile:

File: src/test_helpers/component.cljs (hunk 1 above)
Comment (ilmotta) [reply src/test_helpers/component.cljs:76]: I don't have any particular opinion @ulisesmac. I think the name you chose is consistent with the others, like `is-null`.

**Review by ilmotta — APPROVED** [review]
LGTM :fire:

File: src/quo2/components/inputs/input/component_spec.cljs [hunk 3]
```diff
@@ -0,0 +1,70 @@
+(ns quo2.components.inputs.input.component-spec
+  (:require [quo2.components.inputs.input.view :refer [input]]
```
Comment (J-Son89) [inline src/quo2/components/inputs/input/component_spec.cljs]: I think we want to avoid using refer :)

**Review by J-Son89 — APPROVED** [review]

File: src/quo2/components/inputs/input/component_spec.cljs (hunk 3 above)
Comment (ulisesmac) [reply src/quo2/components/inputs/input/component_spec.cljs]: Solved!

**Comment by ulisesmac** [comment]
This PR has been merged without E2E Tests because it only adds  test for the input component and fixes a visual error on it, but this input is a new component, so it's not used in the app yet.

**Pull request merged**